volatility3/framework/symbols/windows/pdbutil.py

```diff
@@ -226,13 +227,12 @@ def get_guid_from_mz(
             return None
 
         pdb_name = debug_entry.PdbFileName.decode("utf-8").strip("\x00")
+
+        # Let pathlib do the filename extraction. This will likely always be a Windows path though.
+        pdb_name = PureWindowsPath(pdb_name).name
+
         age = debug_entry.Age
-        guid = "{:08x}{:04x}{:04x}{}".format(
-            debug_entry.Signature_Data1,
-            debug_entry.Signature_Data2,
-            debug_entry.Signature_Data3,
-            binascii.hexlify(debug_entry.Signature_Data4).decode("utf-8"),
-        )
+        guid = debug_entry.Signature_String[:32]  # Removes the Age from the GUID
```

Comment:
Am I correct in thinking that `Signature_String` is new to pefile, and so wont be found in older versions? It's probably wise to either update the requirements file with the minimum version, or maybe add a try/except around it?

Reply:
Correct but there's no maximum value set for pefile in the requirements.txt, so anyone installing Vol3 is always getting the newest version anyway.

Might be time to update the minimum versions of dependencies.

Reply:
Yes sorry that's what I mean, setting a minimum version in the requirements given that it would actually break now if there was an older version would probably be helpful.

Reply:
So whilst also requiring the minimum version on pefile, I just also want to check that the old guid values and the new guid values will match if the same file is provided to both the old code and the new code?  Happy to use the new method (and do less manually ourselves), but I want to ensure it returns exactly the same output...

Reply:
As it stands now, the "old code" doesn't actually work at all in this function and probably hasn't since February. I use Rich for tracebacks, so errors are easier to understand but if you look at the ``debug_entry`` local here, you'll see the new pefile structure with 6 fields as opposed to 4, while the current Vol3 code still looks for only the 4 fields.

Current code:
```python
 /home/dcm/dev/volatility3/volatility3/framework/symbols/windows/pdbutil.py:239 in                │
│ get_guid_from_mz                                                                                 │
│                                                                                                  │
│   236 │   │   │   debug_entry.Signature_Data1,                                                   │
│   237 │   │   │   debug_entry.Signature_Data2,                                                   │
│   238 │   │   │   debug_entry.Signature_Data3,                                                   │
│ ❱ 239 │   │   │   binascii.hexlify(debug_entry.Signature_Data4).decode("utf-8"),                 │
│   240 │   │   )                                                                                  │
│   241 │   │   # guid = debug_entry.Signature_String[:32]  # Removes the Age from the GUID        │
│   242 │   │   return guid, age, pdb_name                                                         │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │                  age = 1                                                                     │ │
│ │                  cls = <class 'volatility3.framework.symbols.windows.pdbutil.PDBUtility'>    │ │
│ │              context = <volatility3.framework.contexts.Context object at 0x7f04b1a04fa0>     │ │
│ │           debug_data = <pefile.DebugData object at 0x7f04aa808c10>                           │ │
│ │          debug_entry = <Structure: [CV_INFO_PDB70] 0x15B7F0 0x0 CvSignature: RSDS 0x15B7F4   │ │
│ │                        0x4 Signature_Data1: F9A24309 0x15B7F8 0x8 Signature_Data2: A184      │ │
│ │                        0x15B7FA 0xA Signature_Data3: C71B 0x15B7FC 0xC Signature_Data4: 78   │ │
│ │                        0x15B7FD 0xD Signature_Data5: 85 0x15B7FE 0xE Signature_Data6:        │ │
│ │                        047551E716D6 0x15B804 0x14 Age: 0x1 0x15B808 0x18 PdbFileName:        │ │
│ │                        ntdll.pdb>                                                            │ │
│ │                index = 3                                                                     │ │
│ │                layer = <volatility3.framework.layers.intel.WindowsIntel32e object at         │ │
│ │                        0x7f04aa3bbe80>                                                       │ │
│ │           layer_name = 'layer_name_Process13728_1'                                           │ │
│ │             max_size = 2191360                                                               │ │
│ │               mz_sig = b'MZ'                                                                 │ │
│ │      nt_header_start = 224                                                                   │ │
│ │               offset = 140725005778944                                                       │ │
│ │ optional_header_size = 240                                                                   │ │
│ │             pdb_name = 'ntdll.pdb'                                                           │ │
│ │              pe_data = <pefile.PE object at 0x7f04aa451360>                                  │ │
│ │            pe_header = b'MZ\x90\x00\x03\x00\x00\x00\x04\x00\x00\x00\xff\xff\x00\x00\xb8\x00… │ │
│ │               pe_sig = b'PE'                                                                 │ │
│ │               pefile = <module 'pefile' from                                                 │ │
│ │                        '/home/dcm/dev/volatility3/.venv/lib/python3.10/site-packages/pefile… │ │
│ │        physical_data = b'MZ\x90\x00\x03\x00\x00\x00\x04\x00\x00\x00\xff\xff\x00\x00\xb8\x00… │ │
│ │                 sect = <Structure: [IMAGE_SECTION_HEADER] 0x378 0x0 Name: .reloc 0x380 0x8   │ │
│ │                        Misc: 0x628 0x380 0x8 Misc_PhysicalAddress: 0x628 0x380 0x8           │ │
│ │                        Misc_VirtualSize: 0x628 0x384 0xC VirtualAddress: 0x216000 0x388 0x10 │ │
│ │                        SizeOfRawData: 0x1000 0x38C 0x14 PointerToRawData: 0x20E000 0x390     │ │
│ │                        0x18 PointerToRelocations: 0x0 0x394 0x1C PointerToLinenumbers: 0x0   │ │
│ │                        0x398 0x20 NumberOfRelocations: 0x0 0x39A 0x22 NumberOfLinenumbers:   │ │
│ │                        0x0 0x39C 0x24 Characteristics: 0x42000040>                           │ │
│ │           sizeofHdrs = 4096                                                                  │ │
│ │         virtual_data = b'MZ\x90\x00\x03\x00\x00\x00\x04\x00\x00\x00\xff\xff\x00\x00\xb8\x00… │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: a bytes-like object is required, not 'int'
```

If you're more interested in making sure the GUIDs are correct, here's an example of me using the function from a custom plugin and the output of `symchk.exe` from Windows Debugging Tools to check `ntdll` from the same host. The GUIDs match.

Working code:
```
INFO     volatility3.framework.symbols.windows.pdbconv: Download PDB file...
DEBUG    volatility3.framework.symbols.windows.pdbconv: Attempting to retrieve http://msdl.microsoft.com/download/symbols/ntdll.pdb/F9A24309A184C71B7885047551E716D61/ntdll.pdb
```

Symchk.exe:
```
PS C:\Program Files\Microsoft Visual Studio\2022\Preview> symchk.exe /v C:\Windows\System32\ntdll.dll
[SYMCHK] Searching for symbols to C:\Windows\System32\ntdll.dll in path srv*c:\Symbols*http://msdl.microsoft.com/download/symbols
DBGHELP: Symbol Search Path: srv*c:\Symbols*http://msdl.microsoft.com/download/symbols
[SYMCHK] Using search path "srv*c:\Symbols*http://msdl.microsoft.com/download/symbols"
DBGHELP: No header for C:\Windows\System32\ntdll.dll.  Searching for image on disk
DBGHELP: C:\Windows\System32\ntdll.dll - OK
SYMSRV:  BYINDEX: 0x1
         c:\Symbols*http://msdl.microsoft.com/download/symbols
         ntdll.pdb
         F9A24309A184C71B7885047551E716D61
SYMSRV:  PATH: c:\Symbols\ntdll.pdb\F9A24309A184C71B7885047551E716D61\ntdll.pdb
SYMSRV:  RESULT: 0x00000000
DBGHELP: ntdll - public symbols
        c:\Symbols\ntdll.pdb\F9A24309A184C71B7885047551E716D61\ntdll.pdb
```

Reply:
Sorry I wasn't clear enough.  I'm trying to ensure the API is consistent, so I want to make sure that whatever *did* work with the version available in February is what we return now using the new method.  It sounds like it does, I'm just keen to ensure it's verified that the byte ordering and results end up the same.  If they do that's fine, if they don't we need to think about which ordering is best (whether we make the change and what breaks that might cause, or change the ordering so that the results match the previous version).  This may not be an issue at all, I just want to check before we approve it...  5:)

Reply:
I went back to `pefile==2017.8.1` which is what requirements.txt currently has and it didn't work so I jumped to `2019.4.18` at random and it did. 

```
DEBUG    volatility3.framework.symbols.windows.pdbutil: Using symbol library: ntdll.pdb/F9A24309A184C71B7885047551E716D6-1
```

I dunno how often this function gets used or if at all, but it looks like the minimum requirement for pefile has been wrong since vol3 1.0.0. It just happened to work **at the time**.